src/index.js

```diff
@@ -44,16 +48,112 @@ export default class SHA256 {
 
   static digest64(data) {
     if (data.length==64) {
-      const input = new Uint8Array(staticInstance.ctx.memory.buffer, staticInstance.ctx.input.value, staticInstance.ctx.INPUT_LENGTH);
-      input.set(data);
-      staticInstance.ctx.digest64(staticInstance.ctx.input.value,staticInstance.ctx.output.value);
+      staticInstance.inputArray.set(data);
+      staticInstance.ctx.digest64(staticInstance.wasmInputValue, staticInstance.wasmOutputValue);
       const output = new Uint8Array(32);
-      output.set(new Uint8Array(staticInstance.ctx.memory.buffer, staticInstance.ctx.output.value, 32));
+      output.set(staticInstance.outputArray);
       return output;
     }
     throw new Error("InvalidLengthForDigest64");
   }
 
+  /**
+   * Digest 2 objects, each has 8 properties from h0 to h7.
+   * @returns
+   */
+  static digestObjects(obj1, obj2) {
+    // TODO: expect obj1 and obj2 as Objects
+    objToByteArr(obj1, staticInstance.inputArray, 0);
+    objToByteArr(obj2, staticInstance.inputArray, 32);
+    staticInstance.ctx.digest64(staticInstance.wasmInputValue, staticInstance.wasmOutputValue);
+    return byteArrToObj(staticInstance.outputArray)
+  }
+}
+
+export function objToByteArr(obj, byteArr, offset) {
+  let tmp;
+  for (let index = 0; index < 8; index++) {
```

Comment:
if you unroll this loop, its possible that digestobjects will match the new digest performance

Reply:
Good point! This is the place to go a bit crazy with optimizations, this is a very hot path

Reply:
yeah it saved 2ms in the performance test (so 62ms becomes 60ms). I hope that when we move this bit shifting work to wasm by passing 8 numbers to `inputArray`, we could save some more time.

Reply:
I am sure you will further get 2-3 ms improvement by unrolling the bytes for loop as well. I am not sure if a macro function is available, but that can be used to encapsulate the unrolling for this inner loop.

Reply:
thanks @g11tech , unrolling the other for loop really makes `digestObjects()` the same performance to `digest64()` 👍